### PR TITLE
Refactor storage methods to allow for config passing among controllers.

### PR DESCRIPTION
**Pull request by eschlange**
* Updated each controller to include caching logic.
* Added the configuration service.
* Updated the unit test runner to include config.
* Updated specs to include new logic.
* Updated views to use new config methods.

[#88080168]

### CODE REVIEW

File: src/partials/home.html [hunk 1]
```diff
@@ -22,7 +22,7 @@
                 <span class="days">{{home.cessationDayOffset()}}</span>
                 <span class="suffix">
                     <ng-pluralize count="home.cessationDayOffset()"
-                                  when="{'one': ' day since you quit!','other': ' days since you quit!'}">
+                                  when="{'one': ' day until quit day!','other': ' days until quit day!'}">
```
Comment (michaelwehrley) [inline src/partials/home.html]: I would not want an explanation point there :exclamation: lol

File: src/js/controllers/SetUpController.js [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
+          return configurationService.getCessationDate();
         };
 
         this.setCessationDate = function() {
-            localStorage.setItem(this.CESSATION_DATE_KEY, JSON.stringify(this.cessationDate));
+          configurationService.saveCessationDate(this.cessationDate);
         };
+
+        this.configurationComplete = function() {
+          debugger
```
Comment (michaelwehrley) [inline src/js/controllers/SetUpController.js]: note.

File: src/partials/home.html (hunk 1 above)
Comment (cbrenner04) [inline src/partials/home.html]: I think you mean exclamation mark?

Comment (eschlange) [inline src/partials/home.html]: Just following design artifacts~

File: src/js/controllers/SetUpController.js (hunk 2 above)
Comment (eschlange) [inline src/js/controllers/SetUpController.js]: removing

**Comment by michaelwehrley** [comment]
looks like good refactoring here! :+1:

File: src/partials/home.html (hunk 1 above)
Comment (michaelwehrley) [inline src/partials/home.html]: Dude, you are so right.  Whoops!

**Comment by eschlange** [comment]
@michaelwehrley Thanks, Mike! Should I merge this in?

**Pull request merged**